src/mongoose_client_api/mongoose_client_api_messages.erl

```diff
@@ -137,14 +136,21 @@ encode(Msg, Timestamp) ->
                                      _ ->
                                         []
       end,
-
+    Thread = exml_query:path(Msg, [{element, <<"thread">>}, cdata]), 
+    ThreadParent = exml_query:path(Msg, [{element, <<"thread">>}, {attr, <<"parent">>}]),
+    ThreadAndThreadParentList = case {Thread, ThreadParent} of
+                {undefined, undefined} -> 
+                                        [];
+                {Thread, undefined}    ->
+                                        [{<<"thread">>, Thread}];
```

Comment:
For this case is missed test case, could you please also add test case for this case?